source/tokens/Token.hx

```diff
@@ -0,0 +1,68 @@
+package tokens;
+
+import flixel.util.FlxColor;
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.FlxBasic;
+
+class Token extends FlxSprite
+{
+	public function new(?parameters:TokenParameters)
+		{
+			super();
+
+			if (parameters != null)
+			{
+				init(parameters);
+				
+			}
+		}
+
+	public function init(parameters:TokenParameters)
+		{
+			super.reset(parameters.x, parameters.y);
+	
+			velocity.x = -parameters.speed;
+			makeGraphic(parameters.width, parameters.height);
+			color = FlxColor.YELLOW;
+		}
+	
+		override public function update(elapsed:Float)
+		{
+			if (movedOffScreen())
+			{
+				kill();
+			}
+			super.update(elapsed);
+		}
+	
+		private function movedOffScreen()
+		{
+			return x + width < FlxG.camera.scroll.x;
+		}
+
+		public static function overlapsWithPlayer(player:FlxBasic, scoreToken:Token)
+			{
+				scoreToken.kill();  //Remove token when collide with player
+				//scoreToken.score += 1;    //Add 1 to coin score when player collides
+			}	
+}
```

Comment:
This whole class looks very similar to Obstacle, I think you can probably just inherit from obstacle to get the sort of functionality you want instead of having a bunch of duplicate code.

Reply:
It'd look really similar to DeadlyObstacle, where all it really has to do is set its own color and define the static function that it'll use to handle overlaps.

Reply:
Yea, I realized I could do that but I ended up spending too much time and needed a break. Its changed now though :)